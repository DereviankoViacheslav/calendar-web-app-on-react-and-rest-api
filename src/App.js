import React from 'react';
import './App.css';

function App() {
  return (
    <div>Calendar App</div>
  );
}

export default App;
