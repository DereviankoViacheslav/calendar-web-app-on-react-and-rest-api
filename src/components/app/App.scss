/* common */

* {
  box-sizing: border-box;
}

body {
  padding: 0;
  margin: 0;
  font-family: Roboto;
}

/* header */

.head {
  position: fixed;
  width: 100%;
  background-color: white;
  z-index: 999;
}

.navigate {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  height: 80px;
  padding-left: 120px;
  border-bottom: 1px solid #d5d5d5;
  padding-bottom: 10px;

  &_create {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 50px;
    width: 130px;
    border-radius: 30px;
    border: none;
    outline: none;
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 5px 5px rgba(0, 0, 0, 0.22);
    background-color: white;
    margin-right: 50px;
    font-size: 18px;

    &:hover {
      box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 15px 15px rgba(0, 0, 0, 0.22);
      transition: 1s;
    }
  }

  &_today {
    border: 1px solid grey;
    border-radius: 3px;
    height: 40px;
    width: 90px;
    background-color: transparent;
    outline: none;
    margin-right: 50px;
    font-weight: 700;

    &:hover {
      background-color: #d5d5d5;
      transition: 1s;
    }
  }
}

.navigate__arows {
  display: flex;
  margin-right: 50px;

  &_right {
    background-image: url("https://img.icons8.com/material-sharp/24/000000/forward.png");
    width: 35px;
    height: 35px;
    border-radius: 50%;
    background-position: center;
    background-size: auto;

    &:hover {
      background-color: #D3D3D3;
      transition: 1s;
    }
  }

  &_left {
    background-image: url("https://img.icons8.com/material-sharp/24/000000/forward.png");
    width: 35px;
    height: 35px;
    border-radius: 50%;
    transform: rotate(180deg);
    background-position: center;
    background-size: auto;

    &:hover {
      background-color: #D3D3D3;
      transition: 1s;
    }
  }
}

.navigate__MonthAndYear {
  font-size: 24px;
  font-weight: 500;
}

.plus {
  width: 35px;
  height: 35px;
  margin-right: 10px;
}

.week {
  position: fixed;
  width: 100%;
  display: flex;
  top: 80px;
  z-index: 999;
  padding-left: 60px;
  border-bottom: 1px solid #d5d5d5;
  background-color: white;
}

.day {
  position: relative;
  flex-grow: 1;
  height: 90px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;

  &_nameDay {
    font-size: 18px;
    opacity: 0.5;
  }

  &_numberDay {
    width: 40px;
    height: 40px;
    align-self: center;
    margin-top: 5px;
    font-size: 22px;
    text-align: center;
    line-height: 40px;
    border-radius: 50%;
  }
}

.LittleBorder {
  position: absolute;
  bottom: 0;
  height: 20px;
  width: 100%;
  border-left: 1px solid #d5d5d5;
}

.today {
  background-color: royalblue;
  color: white;
}

/* time-line */

.time-line {
  position: absolute;
  width: 100%;
  height: 2px;
  background-color: red;
}

.time-line:before {
  content: '';
  position: absolute;
  display: block;
  top: -6px;
  left: -8px;
  width: 15px;
  height: 15px;
  border-radius: 50%;
  background-color: red;
}

/* schedule */

.schedule {
  display: flex;
  padding-top: 170px;
}

.sidebar {
  display: flex;
  flex-direction: column;
  color: #70757a;

  :first-child .hour__text {
    display: none;
  }
}

.hour {
  height: 42px;
  width: 55px;

  &__text {
    display: block;
    margin: -9px 0 0 10px;
  }
}

.days {
  display: flex;
  width: 100%;
  padding-left: 5px;
  background: repeating-linear-gradient(180deg,
      #dadce0,
      #dadce0 1px,
      #fff 1px,
      #fff 42px);
}

.column-day {
  position: relative;
  width: 100%;
  border-left: 1px solid #dadce0;
}

/* popup */

.display-none {
  display: none;
}

.popup-layer {
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  z-index: 9999;
  background-color: #89898961;
}

.popup {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  padding: 16px 16px 8px;
  border-radius: 8px;
  text-align: right;
  background-color: #fff;

  &__picker {
    display: flex;
    margin: 45px 0;
    align-items: center;
    position: relative;
  }

  &__color-picker {
    margin: 45px 0 0 45px;
    text-align: left;
  }

  &__color-picker_label-text {
    margin-right: 15px;
  }

  &__btn-close {
    position: absolute;
    right: 10px;
    top: -3px;
  }
}

.event {

  &__name {
    border: none;
    border-bottom: 2px solid #d5d5d5;
    width: 90%;
    font-size: 20px;
    letter-spacing: 2px;
    font-style: italic;
    outline: none;
    margin-top: 30px;
  }

  &__date-start,
  &__date-end {
    padding: 2px;
    margin-left: 18px;
    border: none;
    background-size: contain;
    outline: none;
    width: 150px;
  }

  &__time-start,
  &__time-end {
    border: none;
    background-size: contain;
    outline: none;
    padding: 2px;
    background-color: #fff;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
  }

  &__description {
    border: none;
    width: 90%;
    outline: none;
    resize: none;
  }

  &__btn-save {
    position: absolute;
    right: 5px;
    height: 40px;
    padding-left: 20px;
    padding-right: 20px;
    background: royalblue;
    color: white;
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 5px 5px rgba(0, 0, 0, 0.22);
    border: none;
    border-radius: 5px;
    cursor: pointer;
  }
}

.footer-popup {
  position: relative;
  height: 40px;
  display: flex;
}

.clock {
  width: 25px;
  height: 25px;
}

.line {
  left: 237px;
  width: 8px;
  margin: 0 5px;
  border-top: 2px solid black;
}

.multilineText {
  position: absolute;
  left: 20px;
  width: 30px;
  height: 30px;
}

.close {
  width: 30px;
  height: 30px;
  cursor: pointer;
  margin-top: 10px;
}

.bascket {
  cursor: pointer;
}

.invalid {
  border: 1px solid red;
}

/* event */

.day-event {
  position: absolute;
  min-height: content;
  width: 100%;
  padding: 4px;
  border-radius: 3px;
  background-color: #4183f1;
  font-size: 14px;
  color: white;

  &__title {
    height: 16px;
    overflow: hidden;
  }
}