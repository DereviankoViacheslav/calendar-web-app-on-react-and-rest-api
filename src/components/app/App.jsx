import './App.scss';
import React from 'react';
import Header from '../header'
import ViewSchedule from '../view-schedule';
import Popup from '../popup';
import MockApiService from '../../services/mockApiService';

class App extends React.Component {
  state = {
    dates: [
      '2020-01-27',
      '2020-01-28',
      '2020-01-29',
      '2020-01-30',
      '2020-01-31',
      '2020-02-01',
      '2020-02-02',
    ],
    listEvents: []
  };
  mockApiService = new MockApiService();

  componentDidMount() {
    this.fetchListEvents();
  }

  fetchListEvents = () => {
    this.mockApiService.getEvents()
      .then(listEvents => {
        this.setState({ listEvents });
      });
  }

  // onCreate = (text) => {
  // const patern = {
  //   name: 'Event 1',
  //   createDate: '2020-01-29T07:00:00.000Z',
  //   startDate: '2020-01-29T07:00:00.000Z',
  //   endDate: '2020-01-29T11:15:00.000Z',
  //   description: 'description text',
  //   color: '#f14141',
  // }
  //   this.mockApiService.addEvent(text)
  //     .then(() => this.fetchListEvents());
  // }

  // onDelete = (taskId) => {
  //   this.mockApiService.deleteEvent(taskId)
  //     .then(() => this.fetchListEvents());
  // }

  // onEdit = (taskId) => {
  //   this.mockApiService.editEvent(taskId)
  //     .then(() => this.fetchListEvents());
  // }

  render() {
    const { dates, listEvents } = this.state;

    return (
      <>
        <Header />
        <main className="main">
          <ViewSchedule listEvents={listEvents} dates={dates} />
        </main>
        <Popup />
      </>
    );
  }
}

export default App;