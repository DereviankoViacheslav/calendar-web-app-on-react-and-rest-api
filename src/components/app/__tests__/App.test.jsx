import React from 'react';
import { shallow } from 'enzyme';
import App from '../App'
import moment from 'moment';
import MockApiService from '../../../services/mockApiService';

// const mockApiService = new MockApiService();

jest.mock('../../../services/mockApiService', () => {
  return class {
    getEvents = jest.fn(() => Promise.resolve([]));
    addEvent = jest.fn();
    deleteEvent = jest.fn();
    editEvent = jest.fn();
  };
});

describe('<App />', () => {
  it('should render page', () => {
    const state = {
      firstDayOfWeek: moment('2020-02-12', 'YYYY-MM-DD'),
      listEvents: [],
      dataPoupComponent: null
    };
    const wrappedComponent = shallow(<App />);
    wrappedComponent.setState(state);
    expect(wrappedComponent).toMatchSnapshot();
  });

  it('should display Header', () => {
    const wrappedComponent = shallow(<App />);
    expect(wrappedComponent.find('Header').exists()).toBeTruthy();
  });

  it('should display ViewSchedule', () => {
    const wrappedComponent = shallow(<App />);
    expect(wrappedComponent.find('ViewSchedule').exists()).toBeTruthy();
  });

  it('should not display Popup', () => {
    const wrappedComponent = shallow(<App />);
    expect(wrappedComponent.find('Popup').exists()).toEqual(false);
  });

  it('should display Popup', () => {
    const state = {
      firstDayOfWeek: moment('2020-02-12', 'YYYY-MM-DD'),
      listEvents: [],
      dataPoupComponent: {}
    };
    const wrappedComponent = shallow(<App />);
    wrappedComponent.setState(state);
    expect(wrappedComponent.find('Popup').exists()).toEqual(true);
  });

  it('should called fetchListEvents()', () => {
    // const state = {
    //   firstDayOfWeek: '2020-02-10T00:00:00.000Z',
    //   listEvents: [],
    //   dataPoupComponent: null
    // };
    // const fetchListEvents = jest.fn();
    // const wrappedComponent = shallow(<App />);
    // wrappedComponent.state(state);
    // expect(fetchListEvents).toBeCalled();
  });
  
});