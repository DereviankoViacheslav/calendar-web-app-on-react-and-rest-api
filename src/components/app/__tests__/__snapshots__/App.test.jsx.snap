// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render page 1`] = `
<Fragment>
  <Header
    date={"2020-02-11T22:00:00.000Z"}
    goNextWeek={[Function]}
    goPrevWeek={[Function]}
    goToday={[Function]}
    onShowPopup={[Function]}
  />
  <main
    className="main"
  >
    <ViewSchedule
      date={"2020-02-11T22:00:00.000Z"}
      listEvents={Array []}
      onShowPopup={[Function]}
    />
  </main>
</Fragment>
`;
